src/python/bcc/__init__.py

```diff
@@ -544,6 +581,26 @@ def __delitem__(self, key):
     def __iter__(self):
         return self.tables.__iter__()
 
+    @staticmethod
+    def attach_func(fn, attach_type, attachable_fd=0, flags=0):
+        if not isinstance(fn, BPF.Function):
+            raise Exception("arg 1 must be of type BPF.Function")
+
+        res = lib.bpf_prog_attach(fn.fd, attachable_fd, attach_type, flags)
```

Comment:
Probably we should add bpf_prog_attach/detach2 to libbcc.py to make it explicit?

Reply:
added.